File: Data Structures & Algorithms/guess-number-higher-or-lower/submission-2.py
# The guess API is already defined for you.
# @param num, your guess
# @return -1 if num is higher than the picked number
#          1 if num is lower than the picked number
#          otherwise return 0
# def guess(num: int) -> int:

class Solution:
    def guessNumber(self, n: int) -> int:


        if guess(n) == 0: return n

        high = n
        low = 1

        while True:
            guessNumber = (high+low) // 2
            guessResult = guess(guessNumber)

            if guessResult == 1: # guessNumber < pick
                low = guessNumber
            elif guessResult == -1: # pick > guessNumber
                high = guessNumber
            else:
                return guessNumber

File: Data Structures & Algorithms/duplicate-integer/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */I
    hasDuplicate(nums) {
        let seenMap = new Set()
        for (let num of nums){
            if (seenMap.has(num)) return true
            else seenMap.add(num)
        }
        return false
    }
}
